Cabal/Distribution/Types/ModuleRenaming.hs

```diff
@@ -4,70 +4,71 @@
 module Distribution.Types.ModuleRenaming (
     ModuleRenaming(..),
     defaultRenaming,
-    lookupRenaming,
+    isDefaultRenaming,
 ) where
 
 import Prelude ()
-import Distribution.Compat.Prelude
+import Distribution.Compat.Prelude hiding (empty)
 
 import qualified Distribution.Compat.ReadP as Parse
 import Distribution.Compat.ReadP   ((<++))
-import Distribution.Package
 import Distribution.ModuleName
 import Distribution.Text
 
-import qualified Text.PrettyPrint as Disp
-import Text.PrettyPrint ((<+>), text)
-import qualified Data.Map as Map
-
--- ---------------------------------------------------------------------------
--- Module renaming
+import Text.PrettyPrint
 
 -- | Renaming applied to the modules provided by a package.
 -- The boolean indicates whether or not to also include all of the
 -- original names of modules.  Thus, @ModuleRenaming False []@ is
 -- "don't expose any modules, and @ModuleRenaming True [("Data.Bool", "Bool")]@
 -- is, "expose all modules, but also expose @Data.Bool@ as @Bool@".
+-- If a renaming is omitted you get the 'DefaultRenaming'.
+--
+-- (NB: This is a list not a map so that we can preserve order.)
 --
-data ModuleRenaming = ModuleRenaming Bool [(ModuleName, ModuleName)]
+data ModuleRenaming = ModuleRenaming [(ModuleName, ModuleName)]
+                    | DefaultRenaming
+                    | HidingRenaming [ModuleName]
```

Comment:
Could we get docs on the individual constructors. In particular for each one how it corresponds to the surface syntax. That'll help people connect it to other things they know (or should know).